### PR TITLE
DTSPB-1066: Create ExUI test for solicitor caveat journey

### PR DESCRIPTION
**Pull request by mounikahmcts**
https://tools.hmcts.net/jira/browse/DTSPB-1066

### CODE REVIEW

File: src/test/end-to-end/helpers/PuppeteerHelper.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ class PuppeteerHelper extends Helper {
         if (testConfig.TestForXUI) {
             const tabXPath = `//div[text()='${tabTitle}']`;
             const clickableTab = await helper.page.$x(tabXPath);
-            await clickableTab[0].click();
+            await helper.page.evaluate(el => el.click(), clickableTab[0]);
```
Comment (vinceholt) [inline src/test/end-to-end/helpers/PuppeteerHelper.js:39]: Did we try await I.waitForClickable here then a click?

File: src/test/end-to-end/pages/caseDetails/solicitorApplyCaveat/caveatDetailsTabConfig.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "completedApplication": {
+      "fields": [
+        "Caveat expiry date"
+      ],
+      "dataKeys": [
+        "caveat_expiry_date"
+      ]
+    }
+  }
+
```
Comment (vinceholt) [inline src/test/end-to-end/pages/caseDetails/solicitorApplyCaveat/caveatDetailsTabConfig.json]: add a carriage return at the end of file to remove this red error

File: src/test/end-to-end/pages/caseDetails/solicitorApplyCaveat/notificationsTabConfig.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        "Generated by"
+      ],
+      "dataKeys": [
+        "notification_sent_email",
+        "notification_pdf",
+        "notification_date"
+      ]
+    }
+  }
+
```
Comment (vinceholt) [inline src/test/end-to-end/pages/caseDetails/solicitorApplyCaveat/notificationsTabConfig.json]: carriage return

Comment (vinceholt) [inline src/test/end-to-end/pages/caseDetails/solicitorApplyCaveat/notificationsTabConfig.json]: carriage return

File: src/test/end-to-end/pages/caseDetails/solicitorApplyCaveat/paymentDetailsTabConfig.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    "fields": [
+      "How do you want to pay?",
+      "Fee account number"
+    ],
+    "dataKeys": [
+      "page1_payment_type",
+      "page1_pay_account_num"
+    ]
+  }
+
```
Comment (vinceholt) [inline src/test/end-to-end/pages/caseDetails/solicitorApplyCaveat/paymentDetailsTabConfig.json]: carriage return

File: src/test/end-to-end/helpers/PuppeteerHelper.js (hunk 1 above)
Comment (mounikahmcts) [reply src/test/end-to-end/helpers/PuppeteerHelper.js:39]: The clickableTab is element handle fetched from xpath. so i have updated method to use waitForXpath not waitForClickable

File: src/test/end-to-end/pages/caseDetails/caseDetails.js [hunk 5]
```diff
@@ -43,8 +43,7 @@ module.exports = async function (caseRef, tabConfigFile, dataConfigFile, nextSte
             await I.waitForText(eventSummaryPrefix + dataConfigFile.comment, testConfig.TestTimeToWaitForText || 60);
         }
 
-    } else {
-
+    } else if (dataConfigKeys) {
```
Comment (Rob-Doherty) [inline src/test/end-to-end/pages/caseDetails/caseDetails.js:46]: Is there any case where you might want to handle an `else {}` path? i.e. where `tabConfigFile.tabName !== 'Event History'` and `!dataConfigKeys` ?

**Review by vinceholt — APPROVED** [review]

File: src/test/end-to-end/pages/solicitorApplyCaveat/applyCaveat/page1.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+const commonConfig = require('src/test/end-to-end/pages/common/commonConfig');
+
+module.exports = async function () {
+    const I = this;
+    await I.waitForElement('#solsCaveatEligibility');
+    await I.waitForNavigationToComplete(commonConfig.continueButton);
+};
```
Comment (Rob-Doherty) [inline src/test/end-to-end/pages/solicitorApplyCaveat/applyCaveat/page1.js:9]: is this step actually doing anything other than wait? if not, then maybe the name of "applyCaveat" might be a little confusing, as it sounds like it's actioning something

File: src/test/end-to-end/pages/solicitorApplyCaveat/completeApplication/page2.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+const testConfig = require('src/test/config.js');
+const completeApplicationConfig = require('./completeApplication');
+const commonConfig = require('src/test/end-to-end/pages/common/commonConfig');
+
+module.exports = async function () {
+    const I = this;
+    await I.waitForText(completeApplicationConfig.page2_waitForText, testConfig.TestTimeToWaitForText);
+
+    await I.waitForNavigationToComplete(commonConfig.goButton);
+};
```
Comment (Rob-Doherty) [inline src/test/end-to-end/pages/solicitorApplyCaveat/completeApplication/page2.js:12]: same here, is this step just waiting?

File: src/test/end-to-end/pages/caseDetails/caseDetails.js (hunk 5 above)
Comment (mounikahmcts) [reply src/test/end-to-end/pages/caseDetails/caseDetails.js:46]: For now there is no need of else scenario in the available configs. I prefer to add it when the scenario comes in.

File: src/test/end-to-end/pages/solicitorApplyCaveat/applyCaveat/page1.js (hunk 6 above)
Comment (mounikahmcts) [reply src/test/end-to-end/pages/solicitorApplyCaveat/applyCaveat/page1.js:9]: this page is the first page of applyCaveat event  where it shows some text for the user which is part of div tag with id solsCaveatEligibility. And once the solicitor reads the content he can click on continue to move further in the applying caveat process. We are making sure the first page appears in the process.

File: src/test/end-to-end/pages/solicitorApplyCaveat/completeApplication/page2.js (hunk 7 above)
Comment (mounikahmcts) [reply src/test/end-to-end/pages/solicitorApplyCaveat/completeApplication/page2.js:12]: this is the second page as part of complete application process. We are making sure the second page appears with the required text and make sure solicitor continues on the complete application through this page.

File: src/test/end-to-end/pages/solicitorApplyCaveat/applyCaveat/page1.js (hunk 6 above)
Comment (Rob-Doherty) [reply src/test/end-to-end/pages/solicitorApplyCaveat/applyCaveat/page1.js:9]: ah I see, so the `waitForNavigationToComplete()` step actually performs a click as well as wait, that's where I got thrown. Maybe a small improvement for a future PR but not in the scope of your changes 👍

File: src/test/end-to-end/pages/solicitorApplyCaveat/completeApplication/page2.js (hunk 7 above)
Comment (Rob-Doherty) [reply src/test/end-to-end/pages/solicitorApplyCaveat/completeApplication/page2.js:12]: and same here, the naming threw me. But not in scope. 👍

**Review by Rob-Doherty — APPROVED** [review]
looks good to me 👍

**Pull request merged**